samples/nrf9160/gps/src/main.c

```diff
@@ -37,44 +53,59 @@ static bool           update_terminal;
 static u64_t          fix_timestamp;
 nrf_gnss_data_frame_t last_fix;
 
+K_SEM_DEFINE(lte_ready, 0, 1);
+
 void bsd_recoverable_error_handler(uint32_t error)
 {
 	printf("Err: %lu\n", (unsigned long)error);
 }
 
-static int enable_gps(void)
+static int setup_modem(void)
 {
-	int  at_sock;
-	int  bytes_sent;
-	int  bytes_received;
-	char buf[2];
+	for (int i = 0; i < ARRAY_SIZE(at_commands); i++) {
 
-	at_sock = socket(AF_LTE, 0, NPROTO_AT);
-	if (at_sock < 0) {
-		return -1;
+		if (at_cmd_write(at_commands[i], NULL, 0, NULL) != 0) {
+			return -1;
+		}
 	}
 
-	for (int i = 0; i < ARRAY_SIZE(at_commands); i++) {
-		bytes_sent = send(at_sock, at_commands[i],
-				  strlen(at_commands[i]), 0);
+	return 0;
+}
 
-		if (bytes_sent < 0) {
-			close(at_sock);
+static void wait_for_lte(char *response)
```

Comment:
I think using the link controller module could help for both functions `wait_for_lte` and `activate_lte`.

Reply:
I had a chat with Jan Tore, one of the authors of the link controller, but it doesn't seem like we are able to get it to work with this use-case. So I stick with my current implementation and then we can update it later when we have more time for debugging.